code/modules/power/apc.dm

```diff
@@ -803,38 +803,44 @@ GLOBAL_LIST_INIT(apc_wire_descriptions, list(
 		if(grabber.a_intent == INTENT_GRAB)
 
 			//Synthpack recharge
-			if((grabber.species.flags & IS_SYNTHETIC) && istype(grabber.back, /obj/item/storage/backpack/marine/smartpack))
-				var/obj/item/storage/backpack/marine/smartpack/s_pack = grabber.back
-				if(grabber.action_busy)
-					return
-
-				if(!do_after(grabber, 20, INTERRUPT_ALL, BUSY_ICON_GENERIC))
+			if((grabber.species.flags & IS_SYNTHETIC))
+				if(istype(grabber.gloves, /obj/item/clothing/gloves/synth))
+					var/obj/item/clothing/gloves/synth/bracer = grabber.gloves
+					bracer.handle_apc_charge(grabber, src)
 					return
 
-				playsound(src.loc, 'sound/effects/sparks2.ogg', 25, 1)
+				if(istype(grabber.back, /obj/item/storage/backpack/marine/smartpack))
+					var/obj/item/storage/backpack/marine/smartpack/s_pack = grabber.back
+					if(grabber.action_busy)
+						return
 
-				if(stat & BROKEN)
-					var/datum/effect_system/spark_spread/spark = new()
-					spark.set_up(3, 1, src)
-					spark.start()
-					to_chat(grabber, SPAN_DANGER("The APC's power currents surge eratically, damaging your chassis!"))
-					grabber.apply_damage(10,0, BURN)
-				else if(cell && cell.charge > 0)
-					if(!istype(s_pack))
+					if(!do_after(grabber, 20, INTERRUPT_ALL, BUSY_ICON_GENERIC))
```

Comment:
Same, need to check they still have the pack + optionally that they're still in range of the APC